sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets

```diff
@@ -66,6 +66,16 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
         <Error Condition="'$(_IsFunctionsSdkBuild)' == 'true'" Text="Microsoft.NET.Sdk.Functions package is meant to be used with in-proc function apps. Please remove the reference to this package in isolated function apps."/>
     </Target>
 
+    <PropertyGroup Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
+        <_FunctionsWorkerExecutableFileName >$(TargetName)$(TargetExt)</_FunctionsWorkerExecutableFileName>
+    </PropertyGroup>
+  
+    <PropertyGroup Condition="$(TargetFrameworkIdentifier) != '.NETFramework'">
+        <_FunctionsNativeExecutableExtension Condition="($(RuntimeIdentifier.StartsWith('win')) or $(DefaultAppHostRuntimeIdentifier.StartsWith('win')))">.exe</_FunctionsNativeExecutableExtension>
+        <_FunctionsWorkerExecutableFileName Condition="'$(PublishAot)' != 'true'">$(TargetName).dll</_FunctionsWorkerExecutableFileName>
+        <_FunctionsWorkerExecutableFileName Condition="'$(PublishAot)' == 'true'">$(TargetName)$(_FunctionsNativeExecutableExtension)</_FunctionsWorkerExecutableFileName>
+    </PropertyGroup>
```

Comment:
This can all be removed and replaced with `$(TargetName)$(TargetExt)`

Reply:
No. This produces incorrect value when doing AOT publishing. I just validated it using 
```xml
<Target Name="_PrintFunctionsWorkerExecutableFileName" BeforeTargets="BeforeBuild">
   <Message Text="Printing TargetName TargetExt: $(TargetName)$(TargetExt)" Importance="high" />
</Target>
```
Doing a publish on an app with `<PublishAot>True</PublishAot>`

```bash
shkr  FunctionApp123  ♥ 10:52  dotnet publish -c release -r win-x64 --no-self-contained
MSBuild version 17.8.0+6cdef4241 for .NET
  Determining projects to restore...
  Restored C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj (in 552 ms).
C:\Program Files\dotnet\sdk\8.0.100-rc.2.23502.2\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. S
ee: https://aka.ms/dotnet-support-policy [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
  FunctionApp123 -> C:\Dev\Temp\FunctionApp123\FunctionApp123\bin\release\net8.0\win-x64\FunctionApp123.dll
  Printing TargetName TargetExt: FunctionApp123.dll
  Determining projects to restore...
  Restored C:\Users\shkr\AppData\Local\Temp\zesn1q4n.5fb\WorkerExtensions.csproj (in 527 ms).
  WorkerExtensions -> C:\Users\shkr\AppData\Local\Temp\zesn1q4n.5fb\buildout\Microsoft.Azure.Functions.Worker.Extensions.dll
  Generating native code
C:\NUGET_PACKAGES\microsoft.azure.functions.worker.core\1.15.0\lib\net5.0\Microsoft.Azure.Functions.Worker.Core.dll : warning IL2104: Assembly 'Microsoft.Azure.Functions.Worker.Core' produced trim wa
rnings. For more information see https://aka.ms/dotnet-illink/libraries [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.azure.functions.worker.grpc\1.14.0\lib\net7.0\Microsoft.Azure.Functions.Worker.Grpc.dll : warning IL3053: Assembly 'Microsoft.Azure.Functions.Worker.Grpc' produced AOT ana
lysis warnings. [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.extensions.configuration.binder\5.0.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Binder.dll : warning IL2104: Assembly 'Microsoft.Extensions.Configuration.Binde
r' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.extensions.configuration.binder\5.0.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Binder.dll : warning IL3053: Assembly 'Microsoft.Extensions.Configuration.Binde
r' produced AOT analysis warnings. [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.extensions.dependencyinjection\5.0.0\lib\net5.0\Microsoft.Extensions.DependencyInjection.dll : warning IL3053: Assembly 'Microsoft.Extensions.DependencyInjection' produced
 AOT analysis warnings. [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.extensions.dependencyinjection\5.0.0\lib\net5.0\Microsoft.Extensions.DependencyInjection.dll : warning IL2104: Assembly 'Microsoft.Extensions.DependencyInjection' produced
 trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
D:\a\_work\1\s\src\DotNetWorker.Core\WorkerInformation.cs(17): warning IL3000: Microsoft.Azure.Functions.Worker.WorkerInformation..cctor(): 'System.Reflection.Assembly.Location.get' always returns an
 empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'. [C:\Dev\Temp\FunctionApp123\FunctionApp123\
FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.azure.functions.worker.grpc\1.14.0\lib\net7.0\Microsoft.Azure.Functions.Worker.Grpc.dll : warning IL2104: Assembly 'Microsoft.Azure.Functions.Worker.Grpc' produced trim wa
rnings. For more information see https://aka.ms/dotnet-illink/libraries [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\azure.core\1.10.0\lib\net5.0\Azure.Core.dll : warning IL2104: Assembly 'Azure.Core' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [C:\Dev\T
emp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\azure.core\1.10.0\lib\net5.0\Azure.Core.dll : warning IL3053: Assembly 'Azure.Core' produced AOT analysis warnings. [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
C:\NUGET_PACKAGES\microsoft.azure.functions.worker.core\1.15.0\lib\net5.0\Microsoft.Azure.Functions.Worker.Core.dll : warning IL3053: Assembly 'Microsoft.Azure.Functions.Worker.Core' produced AOT ana
lysis warnings. [C:\Dev\Temp\FunctionApp123\FunctionApp123\FunctionApp123.csproj]
  FunctionApp123 -> C:\Dev\Temp\FunctionApp123\FunctionApp123\bin\release\net8.0\win-x64\publish\
  Determining projects to restore...
  Restored C:\Users\shkr\AppData\Local\Temp\noszv1bb.2xc\WorkerExtensions.csproj (in 401 ms).
  WorkerExtensions -> C:\Users\shkr\AppData\Local\Temp\noszv1bb.2xc\publishout\Microsoft.Azure.Functions.Worker.Extensions.dll
  WorkerExtensions -> C:\Users\shkr\AppData\Local\Temp\noszv1bb.2xc\publishout\
shkr  FunctionApp123  ♥ 10:53  cd C:\Dev\Temp\FunctionApp123\FunctionApp123\bin\release\net8.0\win-x64\publish\
shkr  publish  ♥ 10:53  dir

    Directory: C:\Dev\Temp\FunctionApp123\FunctionApp123\bin\release\net8.0\win-x64\publish

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----          10/24/2023 10:53 AM                .azurefunctions
-a---          10/24/2023 10:53 AM            383 extensions.json
-a---          10/24/2023 10:53 AM       10666496 FunctionApp123.exe
-a---          10/24/2023 10:53 AM       82808832 FunctionApp123.pdb
-a---          10/24/2023 10:53 AM              2 functions.metadata
-a---          10/20/2023  8:41 AM            274 host.json
-a---          10/24/2023 10:53 AM            249 worker.config.json
```
You can see what you suggested prints "Printing TargetName TargetExt: FunctionApp123.dll", but our publish output is an executable.

Reply:
Dug into targets, what is happening is there is a `apphost.exe` that is renamed during copy, so yeah there is no property to check. But I think we can pull this out of an item group. It would be in the `None` item group where identity == `$(AppHostIntermediatePath)`, and the final value we want is the `%(Link)` metadata on that item.

Reply:
``` XML
<Target Name="_PrintFunctionsWorkerExecutableFileName" AfterTargets="AfterBuild">
    <ItemGroup>
      <_FunctionsTarget Include="%(None.Link)" Condition="%(Identity) == $(AppHostIntermediatePath)" />
    </ItemGroup>
    <Message Text="Printing TargetName TargetExt: @(_FunctionsTarget)" Importance="high" />
  </Target>
```

Reply:
Interestingly this exists even with `PublishAot == false`. Wonder why that is?